### PR TITLE
Add and update documentation

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: agents/node/vcxagent-core/package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "demo:faber:pg": "node demo/faber.js --postgresql",
     "demo:alice:sign": "node demo/alice-signature.js",
     "demo:faber:verify": "node demo/faber-verify-signature.js",
-    "test:legacy:integration": "node demo/integration-test.js",
-    "test:legacy:integration:revocation": "REVOCATION=true node demo/integration-test.js",
+    "test:threaded:integration": "node demo/integration-test.js",
```
Comment (mirgee) [inline agents/node/vcxagent-core/package.json:36]: Let me take this opportunity to say I think something like `demo:e2e`, `demo:full` is more accurate and understandable

File: agents/node/vcxagent-core/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# VCX Agent Core
+VCX Agent Core is implementation of simple agent with persistent file storage. It's based on
+[NodeJS Wrapper](../../../wrappers/node) for `aries-vcx` library.
+
+# Try it
+1. First step is to [compile](../../../libvcx) `aries-vcx` library and make it available on your system.
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:6]: > ...make it available on your system.
How?

File: agents/node/vcxagent-core/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+2. You need to have `aries-vcx` compatible agency ready. See more [info](../../../README.md).
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:7]: "`aries-vcx`-compatible" or even better "agency compatible with `aries-vcx`" (or aries-compatible, although that's not strictly true)

File: agents/node/vcxagent-core/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+3. Run `test:threaded:integration`. This will run sample scenario where:Alice and Faber
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:8]: "where Alice and Faber"

File: agents/node/vcxagent-core/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    - establish Aries connection,
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:9]: maybe just esablish connection

File: agents/node/vcxagent-core/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# VCX Agent Core
+VCX Agent Core is implementation of simple agent with persistent file storage. It's based on
+[NodeJS Wrapper](../../../wrappers/node) for `aries-vcx` library.
+
+# Try it
+1. First step is to [compile](../../../libvcx) `aries-vcx` library and make it available on your system.
+2. You need to have `aries-vcx` compatible agency ready. See more [info](../../../README.md).
+3. Run `test:threaded:integration`. This will run sample scenario where:Alice and Faber
+    - establish Aries connection, 
+    - Faber issues credential to Alice
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:10]: "issues a credential"

File: agents/node/vcxagent-core/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    - Faber requests Alice to prove certain information using the credential.
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:11]: Maybe "information about herself", or "prove ownership of certain attributes obtained with the credential"

File: agents/node/vcxagent-cli/readme.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# VCX Agent CLI
+CLI Utility - simple Aries agent with CLI interface. 
+
+# What you need
+As this tool is based on [vcxagent-core](../vcxagent-core), the prerequirements are the same. You
```
Comment (mirgee) [inline agents/node/vcxagent-cli/readme.md:5]: "requirements" or "prerequisites"

File: agents/node/vcxagent-core/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+# Try it
+1. First step is to [compile](../../../libvcx) `aries-vcx` library and make it available on your system.
+2. You need to have `aries-vcx` compatible agency ready. See more [info](../../../README.md).
+3. Run `test:threaded:integration`. This will run sample scenario where:Alice and Faber
+    - establish Aries connection, 
+    - Faber issues credential to Alice
+    - Faber requests Alice to prove certain information using the credential. 
+    
+# Note 
+You can also have look at [vcxagent-cli](../vcxagent-cli) - simple CLI Aries agent based
```
Comment (mirgee) [inline agents/node/vcxagent-core/readme.md:14]: "And did we mention how simple it is?" :) Not sure it is necessary keep mentioning it...

File: agents/node/vcxagent-cli/readme.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+need to:
+- have compiled `aries-vcx` library,
+- have available running mediator agent.
```
Comment (mirgee) [inline agents/node/vcxagent-cli/readme.md:8]: "compiled and installed"
"have mediator agent running."

**Review by mirgee — CHANGES_REQUESTED** [review]
Sorry, I was in nitpicking mood, I guess - might have gone a little overboard :)

**Comment by Patrik-Stas** [comment]
> Sorry, I was in nitpicking mood, I guess - might have gone a little overboard :)

All good, I'll take a look tomorrow :-) Thank you for comments

File: agents/node/vcxagent-core/package.json (hunk 1 above)
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/package.json:36]: good point, I've renamed to plainly `demo`

File: agents/node/vcxagent-core/readme.md (hunk 7 above)
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/readme.md:11]: The second option sounds too technical, I want for
```
Faber requests Alice to prove certain information about herself (using the credential).
```

File: agents/node/vcxagent-core/readme.md (hunk 9 above)
Comment (Patrik-Stas) [reply agents/node/vcxagent-core/readme.md:14]: Yeah the meaning of the "simple" is rather "it does not have many features" or "this is not meant to built on top of for production". But yeah you are right, it's sufficient to say it's using file storage and everyone should be clear :-)

**Pull request merged**